internal/plugin/connectors/http/generic/config.go

```diff
@@ -85,23 +87,45 @@ func newConfig(cfgYAML *ConfigYAML) (*config, error) {
 		cfg.CredentialPatterns[cred] = re
 	}
 
-	// Validate and save header template strings
-	for header, tmplStr := range cfgYAML.Headers {
-		tmpl := newHeaderTemplate(header)
+	cfg.Headers, errs = validateAndSaveTemplateString(cfgYAML.Headers, errs)
+	cfg.QueryParams, errs = validateAndSaveTemplateString(cfgYAML.QueryParams, errs)
+
+	if err := errs.Filter(); err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func validateAndSaveTemplateString(templates map[string]string,
+	errs validation.Errors) (map[string]*template.Template, validation.Errors) {
+	parsedTemplates := make(map[string]*template.Template)
+	// Validate and save template strings
+	for tmplName, tmplStr := range templates {
+		tmpl := newHTTPTemplate(tmplName)
 		// Ignore pointer to receiver returned by Parse(): it's just "tmpl".
 		_, err := tmpl.Parse(tmplStr)
 		if err != nil {
-			errs[header] = fmt.Errorf("invalid header template: %q", err)
+			errs[tmplName] = fmt.Errorf("invalid query param template: %q", err)
```

Comment:
This function is run for both headers and query params so this error shouldn't be used for both.

Reply:
Goooood catch